### DISCUSSION TITLE
(Server?) Option to Disable Inventory Hotkeys.

### DISCUSSION THREAD

**Discussion by MarkCringan** [FakeFishGames/Barotrauma#12766] [Ideas and suggestions]
I understand if this idea is not favorable by many, but I believe it would add to the immersion of many players to have the **option** to do so.

It would be a very welcomed addition if we were given the option to disable inventory hotkeys (on a server side?). This would make things like instantly equipping tools and being able to instantly reload weapons ineffective. 

I like the idea that in order to grab a tool from your inventory or to reload a handheld weapon, there should be more input the player has to do (clicking on the item or dragging in the magazine) to simulate the grabbing / reloading action.

Plus being able to fire any firearm **nonstop** by just having extra ammo in your inventory feels a bit "cheesy".

**Comment by MapleWheels** [comment]
As the person who made Hotkey Reload-Swap, I am telling you this basically not worth the time. The act of instant reload is actually just doing what the player does but in code and the better way to accomplish this is by implementing weapon firing cooldowns. Take a look at the mods `Sabre Armaments` and `Enhanced Armaments` for examples.

If you did restrict this on the server-side "somehow" it would take all of 3 seconds to get around it.

This would also create asymmetrical feel to gameplay where players would be wondering about why they can't swap the item's inventory contents and there's no UI feedback for when the player would be allowed to. All in all this is not worth it, there are other balancing methods available.

**Comment by jigardi** [comment]
Have my vote op!
